seedsource_core/django/seedsource/serializers.py

```diff
@@ -107,3 +112,35 @@ class RegionSerializer(serializers.ModelSerializer):
     class Meta:
         model = Region
         fields = ('name',)
+
+
+class ShareURLSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ShareURL
+        fields = ('configuration', 'version', 'hash', 'created', 'accessed')
+        read_only_fields = ('hash', 'created', 'accessed')
+
+    def create(self, validated_data):
+        configuration = validated_data['configuration']
+        version = validated_data['version']
+        string_to_hash = configuration + str(version)
+        hash_as_bytes = hashlib.sha256(string_to_hash.encode()).digest()
+        hash_as_integer = int.from_bytes(hash_as_bytes, 'big')
+        hash_as_b62_truncated = ""
+        for i in range(8):
+            hash_as_b62_truncated += B62_CHARS[hash_as_integer % 62]
+            hash_as_integer //= 62
+
+        attributes = {
+            'hash': hash_as_b62_truncated,
+            'configuration': configuration,
+            'version': version
+        }
+
+        try:
+            obj = ShareURL.objects.get(hash=hash_as_b62_truncated)
+        except ShareURL.DoesNotExist:
+            obj = ShareURL(**attributes)
+            obj.save()
```

Comment:
This is a good place to use [`get_or_create`](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#get-or-create):

```python
return ShareURL.objects.get_or_create(
    hash=hash_as_b62_truncated, 
    defaults={'configuration': configuration, 'version': version}
)[0]
```